packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -856,7 +856,7 @@ class _CupertinoDialogRenderElement extends RenderObjectElement {
   @override
   void moveRenderObjectChild(RenderObject child, _AlertDialogSections oldSlot, _AlertDialogSections newSlot) {
     if (!allowMoveRenderObjectChild) {
-      super.moveRenderObjectChild(child, oldSlot, newSlot);
+      assert(false);
```

Comment:
From context and from the PR [1] that introduced the `allowMoveRenderObjectChild` flag it seemed like the assumption is that moveRenderObjectChild is never called when allowMoveRenderObjectChild is false, hence this change here.

[1] https://github.com/flutter/flutter/commit/242581418fb9ec308c1b552a08ed93b5eec46a00#diff-4ea2907696de4fecf42f2ca56278a7b1d90c28921b1259d18ae28b1a2553d8e8L413